crate/server/src/middlewares/jwt_auth.rs

```diff
@@ -130,11 +140,21 @@ where
 
         match private_claim {
             Err(_) => Box::pin(async move {
-                jwt_config.jwks.refresh().await?;
-
-                let private_claim = jwt_config
-                    .decode_bearer_header(&identity)
-                    .map(|claim| claim.email);
+                // Refresh jwks
```

Comment:
Maybe give the reason why a refresh would ever be needed? It would justify why not looping on try-decoding/refreshing until you have a winner.

Reply:
Plus, is all the code in-between failure and refresh needed for the retry? Otherwise push it somewhere else.